lib/ohai/plugins/linux/network.rb

```diff
@@ -81,8 +81,16 @@ def check_routing_table(family, iface, default_route_table)
       logger.trace("Plugin Network: Parsing #{line}")
       if /\\/.match?(line)
         parts = line.split("\\")
-        route_dest = parts.shift.strip
+        dest_parts = parts.shift.strip.split("\s", 2)
+        route_dest = dest_parts[0]
         route_endings = parts
```

Comment:
This is pretty confusing way to do this in ruby.

The hidden shift changes parts, and then you assign a pointer of parts to route_endings (everything in ruby is a pointer).

It totally works, but I'd much rather something like this, which I find easier to follow:

```ruby
# If we have multiroute, then the first part will be a normal looking route:
#   default proto ra metric 1024 <other options>
# and then each successive part after that is hops that won't have those options.
# so the first thing we do, is grab that first part, and split it in to the route dest ("default")
# in the above, and then all the options for that route. 
parts = line.split('\\')
route_dest, dest_opts = parts.first.split(nil, 2)
# then all the route endings, generally just hops:
route_endings = parts[1..]
if dest_parts.count > 1
  # your comment here, it's excellent.
  route_endings.map! { |e| e.include?("nexthop") ? "#{dest_opts} #{e}" : e }
end
```